### ISSUE TITLE
Stale operator check may cancel regular operators.

### ISSUE THREAD

**Issue by Luffbee** [pingcap/pd#1714]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
Benchmark the hotspot scheduler.

2. What did you expect to see?
Cluster is health.

3. What did you see instead?
![image](https://user-images.githubusercontent.com/17215435/63906150-62c4d180-ca49-11e9-8308-48fc5a568951.png)
And PD log:
```
[2019/08/29 01:42:37.128 +00:00] [INFO] [operator_controller.go:312] ["add operator"] [region-id=304442] [operator="\"move-hot-write-region {mv peer: store 4 to 5} (kind:region,hot-region, region:304442(4933,2636), createAt:2019-08-29 01:42:37.128199297 +0000 UTC m=+43486.400121199, startAt:0001-01-01 00:00:00 +0000 UTC, currentStep:0, steps:[add learner peer 353377 on store 5, promote learner peer 353377 on store 5 to voter, remove peer on store 4])\""]
[2019/08/29 01:42:37.128 +00:00] [INFO] [operator_controller.go:415] ["send schedule command"] [region-id=304442] [step="add learner peer 353377 on store 5"] [source=create]
[2019/08/29 01:42:37.130 +00:00] [INFO] [cluster.go:370] ["region ConfVer changed"] [region-id=304442] [detail="Add peer:{id:353377 store_id:5 is_learner:true }"] [old-confver=2636] [new-confver=2637]
[2019/08/29 01:42:37.130 +00:00] [INFO] [operator_controller.go:106] ["stale operator"] [region-id=304442] [operator="\"move-hot-write-region {mv peer: store 4 to 5} (kind:region,hot-region, region:304442(4933,2636), createAt:2019-08-29 01:42:37.128199297 +0000 UTC m=+43486.400121199, startAt:2019-08-29 01:42:37.128247691 +0000 UTC m=+43486.400169598, currentStep:0, steps:[add learner peer 353377 on store 5, promote learner peer 353377 on store 5 to voter, remove peer on store 4])\""] [diff=1]
```

4. What version of PD are you using (`pd-server -V`)?
Self modified version (master [14b91e8fbe3](https://github.com/pingcap/pd/tree/14b91e8fbe3c172adcae880e3b858379e3d90eff) merged)

The above problem seems caused by pending learners:
https://github.com/pingcap/pd/blob/14b91e8fbe3c172adcae880e3b858379e3d90eff/server/schedule/operator/operator.go#L228
Pending learners will cause `ConfVer` is changed while the step is unfinished.

The questions here are:
* How to fix the "pending learner" problem?
* Is there any other problems related with the stale operator check?

**Comment by Luffbee** [comment]
#1659

**Comment by Luffbee** [comment]
PR #1715 is trying to fix this.

**Comment by Luffbee** [comment]
Fixed by #1715
